SiteParser.java

```diff
@@ -0,0 +1,307 @@
+package biz.redfrigate.merchantservices.web.main;
+
+import au.com.bytecode.opencsv.CSVWriter;
+import biz.redfrigate.merchantservices.web.servlet.InitParams;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import javax.imageio.ImageIO;
+import javax.ws.rs.core.Response;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.net.*;
+import java.nio.charset.Charset;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static au.com.bytecode.opencsv.CSVWriter.NO_QUOTE_CHARACTER;
+
+/**
+ * Grishko
+ * 01.04.2018
+ */
+public class SiteParser extends ServletContainer {
+
+    private final static String siteUrlForParsing = "https://service.nalog.ru/tabak.do";
+    private final static String captchaFileName = "captchaImage.jpg";
+    private static String date = ""; // example = 29.03.2018 if value "" that use current date
```

Comment:
This is static field, but not final. I guess it's a kind of global constant. 

Looking into code i see this variable value is changing by one request and can be change (or reused) by another concurrent request. 
Also value to this field will be assigned during first request and never be changed after. So it means if your application is working for several days it always will use initial value (first request date). 

I don't think it's expected behaivior.